src/block_cache/block_io.c

```diff
@@ -260,6 +261,75 @@ __wt_blkcache_read(WT_SESSION_IMPL *session, WT_ITEM *buf, WT_PAGE_BLOCK_META *b
     return (ret);
 }
 
+static int
+__read_decrypt(WT_SESSION_IMPL *session, WT_ITEM *in, WT_ITEM *out, const uint8_t *addr,
+  size_t addr_size, bool is_delta)
+{
+    WT_BM *bm;
+    WT_BTREE *btree;
+    WT_DECL_RET;
+    WT_ENCRYPTOR *encryptor;
+
+    btree = S2BT(session);
+    bm = btree->bm;
+    encryptor = btree->kencryptor == NULL ? NULL : btree->kencryptor->encryptor;
+
+    if (encryptor == NULL || encryptor->decrypt == NULL)
+        WT_RET(__blkcache_read_corrupt(
+          session, WT_ERROR, addr, addr_size, "encrypted block for which no decryptor configured"));
+
+    if ((ret = __wt_decrypt(
+           session, encryptor, bm->encrypt_skip(bm, session, is_delta), in, out)) != 0)
+        WT_RET(__blkcache_read_corrupt(session, ret, addr, addr_size, "block decryption failed"));
+
+    return (0);
+}
+
+static int
```

Comment:
Add comment for the function?

Reply:
Thanks, done in d71aeb75b834b222165837479b34010bbaf3ef7e